ibis/expr/operations/numeric.py

```diff
@@ -177,6 +177,24 @@ class Clip(ValueOp):
     output_shape = rlz.shape_like("args")
 
 
+@public
+class ClipLower(ValueOp):
+    arg = rlz.strict_numeric
+    lower = rlz.optional(rlz.strict_numeric)
+
+    output_dtype = rlz.dtype_like("arg")
+    output_shape = rlz.shape_like("args")
+
+
+@public
+class ClipUpper(ValueOp):
```

Comment:
I don't think this should be in ibis core. You can add these nodes in your own codebase and compile them however you'd like.

Reply:
Not related to UDT; please skip.